core/src/autogluon/core/task/base/base_task.py

```diff
@@ -162,3 +163,97 @@ def compile_scheduler_options(
     return scheduler_options
+
+
+# TODO: Migrate TextPrediction to use this version, delete old version
+def compile_scheduler_options_v2(
+        scheduler_options, nthreads_per_trial,
+        ngpus_per_trial, num_trials, time_out, search_strategy=None, search_options=None, checkpoint=None, resume=False, visualizer=None,
+        time_attr=None, reward_attr=None, dist_ip_addrs=None, epochs=None):
+    """
+    Updates a copy of scheduler_options (scheduler-specific options, can be
+    empty) with general options. The result can be passed to __init__ of the
+    scheduler.
+
+    Special role of epochs for HyperbandScheduler: If the search_strategy
+    involves HyperbandScheduler and epochs is given, then this value is
+    copied to scheduler_options['max_t']. Pass epochs for applications
+    where the time_attr is epoch, and epochs is the maximum number of
+    epochs.
+
+    :param scheduler_options:
+    :param search_strategy:
+    :param search_options:
+    :param nthreads_per_trial:
+    :param ngpus_per_trial:
+    :param checkpoint:
+    :param num_trials:
+    :param time_out:
+    :param resume:
+    :param visualizer:
+    :param time_attr:
+    :param reward_attr:
+    :param dist_ip_addrs:
+    :param kwargs:
+    :param epochs: See above. Optional
+    :return: Copy of scheduler_options with updates
+
+    """
+    if scheduler_options is None:
+        scheduler_options = dict()
+    else:
+        assert isinstance(scheduler_options, dict)
+    if dist_ip_addrs is None:
+        dist_ip_addrs = []
+    if search_strategy is None:
+        search_strategy = 'random'
+    assert isinstance(search_strategy, str)
+    if search_options is None:
+        search_options = dict()
+    if visualizer is None:
+        visualizer = 'none'
+    if time_attr is None:
+        time_attr = 'epoch'
+    if reward_attr is None:
+        reward_attr = 'validation_performance'
+    _scheduler_options = {
+        'resource': {
+            'num_cpus': nthreads_per_trial, 'num_gpus': ngpus_per_trial},
+        'searcher': search_strategy,
+        'search_options': search_options,
+        'checkpoint': checkpoint,
+        'resume': resume,
+        'num_trials': num_trials,
+        'time_out': time_out,
+        'reward_attr': reward_attr,
+        'time_attr': time_attr,
+        'visualizer': visualizer,
+        'dist_ip_addrs': dist_ip_addrs,
+    }
+    _scheduler_options.update(copy.copy(scheduler_options))
+
+    searcher = searcher_for_hyperband_strategy.get(_scheduler_options['searcher'])
+    if searcher is not None:
+        _scheduler_options['searcher'] = searcher
+        if epochs is not None:
+            _scheduler_options['max_t'] = epochs
+    required_options = [
+        'resource',
+        'searcher',
+        'search_options',
+        'checkpoint',
+        'resume',
+        'num_trials',
+        'time_out',
+        'reward_attr',
+        'time_attr',
+        'visualizer',
```

Comment:
I currently have it available via both, but only for backwards compatibility purposes. I am happy to restrict it to only HPO if we think there isn't much value add. It currently can be specified via an ag_args_fit kwarg.

Reply:
I think we really need a solid tutorial example of using a visualizer however, since it is hard for me to gauge what users are really gaining from this.